consensus/istanbul/core/roundstate.go

```diff
@@ -464,6 +471,28 @@ func (rs *roundStateImpl) GetProposalVerificationStatus(proposalHash common.Hash
 	return
 }
 
+func (rs *roundStateImpl) SetStateProcessResult(proposalHash common.Hash, stateProcessResult *core2.StateProcessResult) {
+	rs.mu.Lock()
+	defer rs.mu.Unlock()
+
+	if rs.stateProcessResults == nil {
+		rs.stateProcessResults = make(map[common.Hash]*core2.StateProcessResult)
+	}
+
+	rs.stateProcessResults[proposalHash] = stateProcessResult
+}
+
+func (rs *roundStateImpl) GetStateProcessResult(proposalHash common.Hash) (stateProcessResult *core2.StateProcessResult, isCached bool) {
```

Comment:
I think it would be cleaner here to not return the bool `isCached`. Given that `stateProcessResult` is a pointer, lack of state can be communicated by returning nil. Maps in go provide the multi value index operation because in the case of a map with non pointer values you need to be able to determine the difference between the zero value and no value at all, but this is not the case here.